libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/UIDevice+SFHardware.m

```diff
@@ -187,119 +232,81 @@ - (NSNumber *)sfsdk_freeDiskSpace {
+    if ([platform isEqualToString:@"iPad11,3"] || [platform isEqualToString:@"iPad11,4"]) return UIDevice3GiPadAir;  // iPad Air (3rd generation)
+    if ([platform isEqualToString:@"iPad11,1"] || [platform isEqualToString:@"iPad11,2"]) return UIDevice5GiPadMini; // iPad mini (5th generation)
+    if ([platform isEqualToString:@"iPad11,6"] || [platform isEqualToString:@"iPad11,7"]) return UIDevice8GiPad;    // iPad (8th generation)
+    if ([platform isEqualToString:@"iPad12,1"] || [platform isEqualToString:@"iPad12,2"]) return UIDevice9GiPad;    // iPad (9th generation)
+    if ([platform isEqualToString:@"iPad13,1"] || [platform isEqualToString:@"iPad13,2"]) return UIDevice4GiPadAir;  // iPad Air (4th generation)
+    if ([platform isEqualToString:@"iPad13,4"] || [platform isEqualToString:@"iPad13,5"] ||
+        [platform isEqualToString:@"iPad13,6"] || [platform isEqualToString:@"iPad13,7"]) return UIDevice11Inch3GiPadPro; // iPad Pro (11-inch, 3rd generation)
+    if ([platform isEqualToString:@"iPad13,8"] || [platform isEqualToString:@"iPad13,9"] ||
+        [platform isEqualToString:@"iPad13,10"] || [platform isEqualToString:@"iPad13,11"]) return UIDevice5G129InchiPadPro; // iPad Pro (12.9-inch, M1)
+    if ([platform isEqualToString:@"iPad13,16"] || [platform isEqualToString:@"iPad13,17"]) return UIDevice5GiPadAir; // iPad Air (5th generation)
+    if ([platform isEqualToString:@"iPad13,18"] || [platform isEqualToString:@"iPad13,19"]) return UIDevice10GiPad; // iPad (10th generation)
+    if ([platform isEqualToString:@"iPad14,1"] || [platform isEqualToString:@"iPad14,2"]) return UIDevice6GiPadMini; // iPad mini (6th generation)
+    if ([platform isEqualToString:@"iPad14,3"] || [platform isEqualToString:@"iPad14,4"]) return UIDevice6G129InchiPadPro; // iPad Pro (12.9-inch, 6th generation)
+    if ([platform isEqualToString:@"iPad14,5"] || [platform isEqualToString:@"iPad14,6"]) return UIDevice11Inch4GiPadPro; // iPad Pro (11-inch, 4th generation)
+ 
+    // Apple TVs
+    if ([platform isEqualToString:@"AppleTV5,3"]) return UIDeviceAppleTV4; // Apple TV HD
+    if ([platform isEqualToString:@"AppleTV6,2"]) return UIDeviceAppleTV4k; // Apple TV 4K (1st generation)
+    if ([platform isEqualToString:@"AppleTV11,1"]) return UIDeviceAppleTV4k2G; // Apple TV 4K (2nd generation)
+    if ([platform isEqualToString:@"AppleTV14,1"]) return UIDeviceAppleTV4k3G; // Apple TV 4K (3rd generation)
+
+    // Simulators
     #if !TARGET_OS_VISION
-        if ([platform hasSuffix:@"86"] || [platform isEqual:@"x86_64"]) {
-            BOOL smallerScreen = [[UIScreen mainScreen] bounds].size.width < 768;
-            return smallerScreen ? UIDeviceSimulatoriPhone : UIDeviceSimulatoriPad;
-        }
+    UIDevice *device = [UIDevice currentDevice];
+    if (device.userInterfaceIdiom == UIUserInterfaceIdiomPhone) {
```

Comment:
This reads to me like it could also be an unknown iPhone instead of a simulator?

Reply:
could potentially use this to replace the architecture check https://github.com/forcedotcom/SalesforceMobileSDK-iOS/blob/40de3762b541473a55685ff28013928a1cd30b9d/libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/UIDevice%2BSFHardware.m#L457